### PR TITLE
Add metadata option to .manifest

### PR DESCRIPTION
**Pull request by marensws**
fixes: #4289

### CODE REVIEW

File: bundle/bundle.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func (m Manifest) Copy() Manifest {
 
 func (m Manifest) String() string {
 	m.Init()
-	return fmt.Sprintf("<revision: %q, roots: %v, wasm: %+v>", m.Revision, *m.Roots, m.WasmResolvers)
+	return fmt.Sprintf("<revision: %q, roots: %v, wasm: %+v, metadata: %x>", m.Revision, *m.Roots, m.WasmResolvers, m.Metadata)
```
Comment (johanneslarsson) [inline bundle/bundle.go:183]: Change `%x`  to `+%v`

Comment (marensws) [reply bundle/bundle.go]: Done

**Review by johanneslarsson — APPROVED** [review]
LGTM

File: compile/compile.go [hunk 2]
```diff
@@ -180,6 +181,12 @@ func (c *Compiler) WithCapabilities(capabilities *ast.Capabilities) *Compiler {
 	return c
 }
 
+//WithMetadata sets the additional data to be included in .manifest
```
Comment (srenatus) [inline compile/compile.go:184]: [nit] 
```suggestion
// WithMetadata sets the additional data to be included in .manifest
```

**Review by srenatus — APPROVED** [review]
LGTM thanks a lot. There's a whitespace nitpick, but please ignore that. Let's merge this.

**Pull request merged**